apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+import { useMemo } from 'react';
+
+import CoinBalance from './coin-balance';
+import IconLink from './icon-link';
+import FaucetMessageInfo from '_app/shared/faucet/message-info';
+import FaucetRequestButton from '_app/shared/faucet/request-button';
+import AccountAddress from '_components/account-address';
+import Alert from '_components/alert';
+import Loading from '_components/loading';
+import RecentTransactions from '_components/transactions-card/RecentTransactions';
+import { SuiIcons } from '_font-icons/output/sui-icons';
+import { useAppSelector, useObjectsState } from '_hooks';
+import { accountAggregateBalancesSelector } from '_redux/slices/account';
+import { GAS_TYPE_ARG, Coin } from '_redux/slices/sui-objects/Coin';
+
+import st from './TokensPage.module.scss';
+
+type TokenDetailsProps = {
+    coinType?: string;
+};
+
+const emptyWalletDescription = (
+    <div className={st.emptyWalletDescription}>
+        To conduct transactions on the Sui network, you need SUI in your wallet.
+    </div>
+);
+
+type TokensProps = {
+    allCoinTypes: string[];
+    suiBalance: bigint;
+    balances: Record<string, bigint>;
+    loading: boolean;
+};
+
+function MyTokens({
```

Comment:
I would move this to a separate file and probably use store for data - but if we can do it in another PR